protocol/x/clob/memclob/memclob.go

```diff
@@ -935,13 +936,14 @@ func (m *MemClobPriceTimePriority) ReplayOperations(
 				continue
 			}
 
-			if m.operationsToPropose.IsOrderRemovalInOperationsQueue(statefulOrderPlacement.Order.OrderId) {
-				// If an order removal for this order is found in the ops queue, we have encountered an error. A
-				// stateful order should not be on the book if it hasn't been replayed yet which means it should not
-				// be possible for an order removal to precede a stateful order placement.
+			// Log an error if the order was already placed. We should not see duplicate Preexisting Stateful Orders
+			// and Order Removals (which, when replayed, can result in placements) should never precede a Preexisting
+			// Stateful Order operation.
+			if _, found := placedStatefulOrderIds[*orderId]; found {
```

Comment:
If order removals should never precede a preexisting stateful order operation, should we add this check to ValidateBasic for the `MsgProposedOperation`? We would catch this before we enter DeliverTx.

Reply:
Preexisting stateful order operations are not included in the operations queue that is proposed in MsgProposedOperations. Preexisting stateful orders are kept only as internal operations and are only used during the ReplayOperations step. So I don't think we could add validation for this

Reply:
I was actually wrong about this. Preexisting stateful orders are not proposed, however we should be able to validate that a stateful order is not present as a taker in a match after an Order Removal in the queue. This will always cause the set of operations to fail execution in DeliverTx. Will cut an issue for this, thanks for suggesting @jonfung-dydx